File: app/src/main/java/nl/rsdt/japp/jotial/maps/kml/KmlLocation.java
package nl.rsdt.japp.jotial.maps.kml;

import java.util.ArrayList;
import java.util.List;

public class KmlLocation {
    public final double lat;
    public final double lon;
    public final double alt;

    public KmlLocation(Double lat, Double lon, Double alt) {
        this.lat = lat;
        this.lon = lon;
        this.alt = alt;
    }

    public static List<KmlLocation> readCoordinates(String coordinates) {
        List<KmlLocation> result = new ArrayList<>();
        for (String c : coordinates.split("\n")) {
            c = c.trim();
            String[] coordinate = c.split(",");
            if (coordinate.length != 3) throw new RuntimeException("string is not an coordinate");
            result.add(new KmlLocation(
                    Double.valueOf(coordinate[0]),
                    Double.valueOf(coordinate[1]),
                    Double.valueOf(coordinate[2])
            ));
        }
        return result;
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/maps/kml/KmlLocation.kt
package nl.rsdt.japp.jotial.maps.kml

import java.util.*

class KmlLocation(val lon: Double?, val lat: Double?, val alt: Double?) {
    companion object {

        fun readCoordinates(coordinates: String): List<KmlLocation> {
            val result = ArrayList<KmlLocation>()
            for (c in coordinates.split("\n".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()) {
                c = c.trim { it <= ' ' }
                val coordinate = c.split(",".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
                if (coordinate.size == 3) {
                    result.add(KmlLocation(
                            java.lang.Double.valueOf(coordinate[0]),
                            java.lang.Double.valueOf(coordinate[1]),
                            java.lang.Double.valueOf(coordinate[2])
                    ))
                }
            }
            return result
        }
    }
}
